WolvenKit.Common/Tools/HashHelper.cs

```diff
@@ -16,7 +17,13 @@ internal static class HashHelper
     /// </summary>
     /// <param name="resourcePath"></param>
     /// <returns></returns>
-    public static ulong CalculateDepotPathHash(ResourcePath resourcePath) => CalculateDepotPathHash(resourcePath.GetResolvedText());
-
+    public static ulong CalculateDepotPathHash(ResourcePath resourcePath)
+    {
+        if (resourcePath.GetResolvedText() is string str && str != "")
```

Comment:
Whats that for?
It only adds overhead to the function call. Since `GetResolvedText()` will lookup the hash of `ResourcePath` in our lookup table and if it's found you do the FNV1A64 on it which will result in the same hash as you started the lookup with.

So it should be one of these:
```cs#
public static ulong CalculateDepotPathHash(ResourcePath resourcePath) => resourcePath.GetRedHash();
or
public static ulong CalculateDepotPathHash(ResourcePath resourcePath) => resourcePath;
```

Or did I miss something?

Reply:
That's probably me not trusting implicit casts anymore and being paranoid and/or confused. You're totally right.